R/arg.R

```diff
@@ -47,9 +46,31 @@ arg_match <- function(arg, values = NULL) {
   }
 
   arg <- arg[[1]]
+  arg_match0(arg, values, arg_nm)
+}
+
+#' @description
+#' `arg_match0()` is a bare-bones version if performance is at a premium.
+#' It requires a string as `arg` and explicit `values`.
+#' @param arg_nm The label to be used for `arg` in error messages.
+#' @rdname arg_match
+#' @export
+#' @examples
+#'
+#' # Use the bare-bones version with explicit values for speed:
+#' arg_match0("bar", c("foo", "bar", "baz"))
+arg_match0 <- function(arg, values, arg_nm = as_label(substitute(arg))) {
+  # Use primitive checks to avoid function calls
+  if (!.Call(rlang_is_string, arg, NULL)) {
+    abort(paste0(chr_quoted(arg_nm), " must be a string."))
+  }
+  if (!.Call(rlang_is_character, arg, NULL)) {
```

Comment:
Is this tested? This should be `values`. If we're bypassing the public API this definitely needs tests (but I'd prefer to use the proper function calls as mentioned above).